api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db;
+
+import java.util.List;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+
+/**
+ * This interface defines database methods for diagnosis objects.
+ *
+ * @since 2.2
+ */
+public interface DiagnosisDAO {
+	
+	/**
+	 * Saves the diagnosis
+	 *
+	 * @param diagnosis the diagnosis to save
+	 * @return the saved diagnosis
+	 * @throws DAOException exception thrown if error occurs while saving the diagnosis
+	 */
+	Diagnosis saveDiagnosis(Diagnosis diagnosis) throws DAOException;
+	
+	/**
+	 * Gets a diagnosis
```

Comment:
Gets a diagnosis is not enough description

Reply:
Updating this